### DISCUSSION TITLE
Possible integration with Interception?

### DISCUSSION THREAD

**Discussion by tanoem** [AntiMicroX/antimicrox#606] [Ideas]
I have recently started using this for my Belkin n50 as the original software is quite dated and wouldn't auto-switch profiles with 64-bit games. It has definitely been working great so far. One idea that came up that maybe it might be worth investigating the use of the Interception API available here: http://www.oblita.com/interception

I have used it myself to write a simple key mapping program for my Razer Nostromo since otherwise I would have to use the Razer Synapse 2 software which is awful. What I put together was very simple though with just simple config files and not the nice user interface of AntiMicroX which would be better suited for integrating something like this.

That said, the usage of this has upsides and downsides that are as follows:

Upsides
* Allows the remapping of just about any device that shows up as another keyboard. Someone can take a simple numeric keypad and remap it to perform other functions.
* Keyboard devices can be uniquely identified through the `interception_get_hardware_id` function, so the main keyboard will work normally and extra devices attached to a PC can be configured their own way.
* It has an LGPL license for non-commercial use so it should be able to be integrated just fine into AntiMicroX.

Downsides
* It requires the installation of a filter driver in Windows which might be annoying to deal with.
* The driver itself it closed source. The binary can be distributed freely for non-commercial purposes, but some people may not like the closed source nature of it
* Adding mapping of keyboard devices may be a bit of a paradigm shift and require a lot of rework of AntiMicroX in some spots.

I totally understand if this is something that gets shot down, but I figured it wouldn't hurt to bring up. Adding the ability to configure keyboard devices in addition to joystick devices would certainly expand the possibilities.

**Comment by pktiuk** [comment]
Apart from the issues you already mentioned there is also another one:  
Interception is supported only by Windows.
